File: Leetcode/typescript/frequency-sort.ts
function frequencySort(nums: number[]): number[] {
  let freqMap = new Map();
  nums = nums.sort((a, b) => b - 1);
  for (const num of nums) {
    freqMap.set(num, (freqMap.get(num) || 0) + 1);
  }
  nums.sort((a, b) => {
    let freqA = freqMap.get(a) || 0;
    let freqB = freqMap.get(b) || 0;

    if (freqA === freqB) {
      return b - a;
    } else {
      return freqA - freqB;
    }
  });
  return nums;
}

File: Leetcode/python/frequency-sort.py
from typing import List

class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        freqMap = {}

        for num in nums:
            freqMap[num] = freqMap.get(num, 0) + 1

        res = sorted(nums, key=lambda x: (freqMap[x], -x))
        return res
